### ISSUE TITLE
Convert EyeLocationData to EyeHeightProperty

### ISSUE THREAD

**Issue by DDoS** [SpongePowered/SpongeAPI#816]
There's an issue with eye location being dependent on the entity location. It's impossible to calculate the eye location with out it. Unless the entity location can be passed to the new instance in the builder you have to use a default value that isn't very relevant (such as Vector3d.ZERO). You can try and calculate it from the entity location plus the eye height , but if this second value never gets set it is impossible to do. 

Modifying this value is also a problem since it's used client side.

Basically I'm suggesting we convert this to an eye height property. I've already working started on it [here](https://github.com/SpongePowered/SpongeAPI/blob/feature/eye-height-property/src/main/java/org/spongepowered/api/data/property/entity/EyeHeightProperty.java), but right now I'm stuck on properties not being implemented.

**Comment by ryantheleach** [comment]
Oh. I always assumed that eye location was relative to the entities location. Does morph change the eye x/z relative to the entity or just the eye height?

**Comment by simon816** [comment]
Thing is, properties are immutable and cannot be changed, but the eye height can. For example if a player crouches.

**Comment by ryantheleach** [comment]
Can not be changed by plugins, or can not be changed ever?

**Comment by DDoS** [comment]
Well we need something that cannot be modified by a plugin, but which can change with the entity state. Basically it needs to be read-only.

We can also replace eye location by offset, but eye height is only an offset on y, so it's not that useful.

**Comment by simon816** [comment]
Can't be changed at all.
An extract from [`DataHolder#getProperty`](https://github.com/SpongePowered/SpongeAPI/blob/master/src/main/java/org/spongepowered/api/data/DataHolder.java#L45) javadocs
>
```
{@link Property}s can define various immutable information about a
     * {@link DataHolder} that is dependent on the instance of the holder.
     * As {@link Property}s cannot be changed, the {@link DataHolder} can
     * not change the information about it's own properties either.
```

**Comment by DDoS** [comment]
So what can we use instead?

**Comment by gabizou** [comment]
@DDoS you can certainly use a `Property`. @simon816 Those docs probably need more explaining, because let's put it this way: I get a `DamageAbsorptionProperty` because I have a helmet `ItemStack`. If I unequip the `ItemStack` and change it to a stick, obviously the property will not apply. `Property` objects are supposed to be properties of a `DataHolder` at that very instance of time. It may change the next tick due to whatever reason, but the fact is that at the time of code execution, a `DataHolder` will have certain "properties" that can be retrieved and "parsed" for information.

**Comment by gabizou** [comment]
Completed in 84442448b3bc2632765c3ebec0d7e0fd7f2256e9
